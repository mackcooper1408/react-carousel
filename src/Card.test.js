import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

describe("Card component tests", function () {
  it("renders correctly", function () {
    const { container, debug } = render(
      <Card 
        caption="caption-test"
        src="src-test"
        currNum={2}
        totalNum={10} />
    );
  });
  it("matches snapshot", function () {
    const { container, debug } = render(
      <Card 
        caption="caption-test"
        src="src-test"
        currNum={2}
        totalNum={10} />
    );
    debug();
    expect(container).toMatchSnapshot();
  });
  it("has correct src and alt", function () {
    const { container } = render(
      <Card 
        caption="caption-test"
        src="src-test"
        currNum={2}
        totalNum={10} />
    );
    const img = container.querySelector(".Card-image")
    expect(img.getAttribute("src")).toEqual("src-test");
    expect(img.getAttribute("alt")).toEqual("caption-test");
  });
  it("displays count correctly", function () {
    const { container } = render(
      <Card 
        caption="caption-test"
        src="src-test"
        currNum={2}
        totalNum={10} />
    );
    const count = container.querySelector(".Card-small")
    expect(count).toContainHTML("2 of 10");
  });

});