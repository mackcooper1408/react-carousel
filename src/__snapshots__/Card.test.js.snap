// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component tests matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      caption-test
    </h4>
    <img
      alt="caption-test"
      class="Card-image"
      src="src-test"
    />
    <small
      class="Card-small"
    >
      Image 
      2
       of 
      10
      .
    </small>
  </div>
</div>
`;
